test/new-e2e/tests/cws/fargate/fargate_test.go

```diff
@@ -0,0 +1,471 @@
+	return ddCWSInstrumentationConfig{
+		sdkconfig.New(e.Ctx, DDCWSInstrumentationNamespace),
+		e,
+	}
+}
+
+func (c *ddCWSInstrumentationConfig) getFullImagePath() string {
+	return c.GetStringWithDefault(c.CWSInstrumentationConfig, DDCWSInstrumentationFullImagePathParamName, "docker.io/datadog/cws-instrumentation-dev:safchain-custom-cws-inst")
+}
+
+type ECSFargateSuite struct {
+	suite.Suite
+	ctx       context.Context
+	stackName string
+	testID    string
+
+	apiClient      *e2elib.APIClient
+	ddHostname     string
+	ecsClusterArn  string
+	ecsClusterName string
+	fgTaskDefArn   string
+}
+
+func TestECSFargate(t *testing.T) {
+	suite.Run(t, &ECSFargateSuite{
+		ctx:       context.Background(),
+		stackName: "cws-tests-ecs-fg",
+		testID:    e2elib.RandomString(4),
+	})
+}
+
+func (s *ECSFargateSuite) SetupSuite() {
+	s.apiClient = e2elib.NewAPIClient()
+
+	ruleDefs := []*e2elib.TestRuleDefinition{
+		{
+			ID:         execRuleID,
+			Expression: fmt.Sprintf(`exec.file.path == \"%s\"`, execFilePath),
+		},
+		{
+			ID:         openRuleID,
+			Expression: fmt.Sprintf(`open.file.path == \"%s\"`, openFilePath),
+		},
+	}
+	selftestsPolicy, err := e2elib.GetPolicyContent(ruleDefs)
+	s.Require().NoError(err)
+
+	_, result, err := infra.GetStackManager().GetStack(s.ctx, s.stackName, nil, func(ctx *pulumi.Context) error {
+		ddHostname := fmt.Sprintf("%s-%s", ddHostnamePrefix, s.testID)
+		awsEnv, err := awsResources.NewEnvironment(ctx)
+		if err != nil {
+			return err
+		}
+		cwsCfg := newCWSInstrumentationConfig(awsEnv.CommonEnvironment)
+
+		// Create cluster
+		ecsCluster, err := ecsResources.CreateEcsCluster(awsEnv, "cws-cluster")
+		if err != nil {
+			return err
+		}
+
+		// Export cluster’s properties
+		ctx.Export(ecsClusterNameKey, ecsCluster.Name)
+		ctx.Export(ecsClusterArnKey, ecsCluster.Arn)
+
+		// Associate Fargate capacity provider to the cluster
+		_, err = ecsResources.NewClusterCapacityProvider(awsEnv, "cws-cluster-capacity-provider", ecsCluster.Name, pulumi.StringArray{pulumi.String("FARGATE")})
+		if err != nil {
+			return err
+		}
+
+		// Setup agent API key
+		apiKeyParam, err := ssm.NewParameter(ctx, awsEnv.Namer.ResourceName("agent-apikey"), &ssm.ParameterArgs{
+			Name:  awsEnv.CommonNamer.DisplayName(1011, pulumi.String("agent-apikey")),
+			Type:  ssm.ParameterTypeSecureString,
+			Value: awsEnv.AgentAPIKey(),
+		}, awsEnv.WithProviders(configCommon.ProviderAWS, configCommon.ProviderAWSX))
+		if err != nil {
+			return err
+		}
+
+		// Create task definition
+		taskDef, err := ecsx.NewFargateTaskDefinition(ctx, "cws-task", &ecsx.FargateTaskDefinitionArgs{
+			Containers: map[string]ecsx.TaskDefinitionContainerDefinitionArgs{
+				"datadog-agent": {
+					Cpu:   pulumi.IntPtr(0),
+					Name:  pulumi.String("datadog-agent"),
+					Image: pulumi.String(agentComp.DockerAgentFullImagePath(awsEnv.CommonEnvironment, "docker.io/datadog/agent-dev", "safchain-custom-cws-inst-py3")),
```

Comment:
I changed the default value of both the agent and cws-instrumentation to the latest rc.
The issue here is that both images must be kept in sync to avoid breaking the protocol between the cws-instrumentation tracer and the agent. Once the development of the cws-instrumentation tracer is more stable we'll be able to avoid pinning both images to a specific release.